### PR TITLE
feat!: update multiformats, publish as ESM

### PR DESCRIPTION
**Pull request by achingbrain**
Updates multiformats module to 10.x.x.  This module is now published as ESM-only so I had to update `tsconfig.json` to do the same here otherwise this module gets transpiled to CJS which then explodes at runtime when trying to import the ESM-only multiformats.

I removed ts-jest as I just couldn't get it to work with ESM-only modules, instead using babel's `@babel/preset-typescript` to compile ts to js.

There were also two snapshots directories - `__snapshots__` and `__SNAPSHOTS__` in this repo which wreak havoc on case-insensitive filesystems like MacOS, hopefully there's only one directory now.

### CODE REVIEW

**Comment by bshambaugh** [comment]
Hey @achingbrain , thanks for looking at this and greatly improving the quality of this repository. As a status update, I just pulled this PR to my local machine and ran it through the linter. As usual, at least for me, I get ESM and CJS complaints for code bases. I'll try to update here as I fiddle around a bit more to be confident what needs to be there. Sergey, over at Ceramic, recently helped me clean up nist-weierstrauss. I'll go over to that repo for inspiration, snyc with you, etc...and get this done..

```
:~/Documents/exploringCIDs/elliptic-piece/did-key-creator$ npm run lint

> did-key-creator@0.3.2 lint
> ./node_modules/.bin/eslint ./src --ext .js,.jsx,.ts,.tsx


Oops! Something went wrong! :(

ESLint: 7.23.0

Error [ERR_REQUIRE_ESM]: require() of ES Module /home/ubuntu/Documents/exploringCIDs/elliptic-piece/did-key-creator/.eslintrc.js from /home/ubuntu/Documents/exploringCIDs/elliptic-piece/did-key-creator/node_modules/@eslint/eslintrc/lib/config-array-factory.js not supported.
.eslintrc.js is treated as an ES module file as it is a .js file whose nearest parent package.json contains "type": "module" which declares all .js files in that package scope as ES modules.
Instead rename .eslintrc.js to end in .cjs, change the requiring code to use dynamic import() which is available in all CommonJS modules, or change "type": "module" to "type": "commonjs" in /home/ubuntu/Documents/exploringCIDs/elliptic-piece/did-key-creator/package.json to treat all .js files as CommonJS (using .mjs for all ES modules instead).

    at module.exports (/home/ubuntu/Documents/exploringCIDs/elliptic-piece/did-key-creator/node_modules/import-fresh/index.js:32:59)
    at loadJSConfigFile (/home/ubuntu/Documents/exploringCIDs/elliptic-piece/did-key-creator/node_modules/@eslint/eslintrc/lib/config-array-factory.js:225:16)
    at loadConfigFile (/home/ubuntu/Documents/exploringCIDs/elliptic-piece/did-key-creator/node_modules/@eslint/eslintrc/lib/config-array-factory.js:309:20)
    at ConfigArrayFactory.loadInDirectory (/home/ubuntu/Documents/exploringCIDs/elliptic-piece/did-key-creator/node_modules/@eslint/eslintrc/lib/config-array-factory.js:502:34)
    at CascadingConfigArrayFactory._loadConfigInAncestors (/home/ubuntu/Documents/exploringCIDs/elliptic-piece/did-key-creator/node_modules/@eslint/eslintrc/lib/cascading-config-array-factory.js:379:46)
    at CascadingConfigArrayFactory._loadConfigInAncestors (/home/ubuntu/Documents/exploringCIDs/elliptic-piece/did-key-creator/node_modules/@eslint/eslintrc/lib/cascading-config-array-factory.js:398:20)
    at CascadingConfigArrayFactory.getConfigArrayForFile (/home/ubuntu/Documents/exploringCIDs/elliptic-piece/did-key-creator/node_modules/@eslint/eslintrc/lib/cascading-config-array-factory.js:300:18)
    at FileEnumerator._iterateFilesRecursive (/home/ubuntu/Documents/exploringCIDs/elliptic-piece/did-key-creator/node_modules/eslint/lib/cli-engine/file-enumerator.js:445:49)
    at _iterateFilesRecursive.next (<anonymous>)
    at FileEnumerator.iterateFiles (/home/ubuntu/Documents/exploringCIDs/elliptic-piece/did-key-creator/node_modules/eslint/lib/cli-engine/file-enumerator.js:296:49)
    at iterateFiles.next (<anonymous>)
    at CLIEngine.executeOnFiles (/home/ubuntu/Documents/exploringCIDs/elliptic-piece/did-key-creator/node_modules/eslint/lib/cli-engine/cli-engine.js:770:48)
    at ESLint.lintFiles (/home/ubuntu/Documents/exploringCIDs/elliptic-piece/did-key-creator/node_modules/eslint/lib/eslint/eslint.js:530:23)
    at Object.execute (/home/ubuntu/Documents/exploringCIDs/elliptic-piece/did-key-creator/node_modules/eslint/lib/cli.js:299:36)
    at main (/home/ubuntu/Documents/exploringCIDs/elliptic-piece/did-key-creator/node_modules/eslint/bin/eslint.js:142:52)
    at Object.<anonymous> (/home/ubuntu/Documents/exploringCIDs/elliptic-piece/did-key-creator/node_modules/eslint/bin/eslint.js:146:2)

```

**Comment by bshambaugh** [comment]
hmm.. as a note to myself..it may be one in the same problem... how to express ESM may just be something I need I learn about.
compare:
https://github.com/bshambaugh/nist-weierstrauss/issues/2

**Comment by bshambaugh** [comment]
fwiw, here are the links I am looking at in order to understand this problem:

https://bobbyhadz.com/blog/javascript-error-err-require-esm-require-of-es-module-not-supported

https://github.com/eslint/eslint/issues/12319

https://stackoverflow.com/questions/69081410/error-err-require-esm-require-of-es-module-not-supported

https://answers.netlify.com/t/error-err-require-esm-require-of-es-module/53937

**Comment by bshambaugh** [comment]
I created a parallel branch [1] to test this out with some of the configuration that came from https://github.com/bshambaugh/nist-weierstrauss . 

[1] https://github.com/bshambaugh/did-key-creator/tree/scratch-main-pr4

```
I am still getting the error like above:
Error [ERR_REQUIRE_ESM]: require() of ES Module /home/ubuntu/Documents/exploringCIDs/elliptic-piece/did-key-creator/.eslintrc.js from /home/ubuntu/Documents/exploringCIDs/elliptic-piece/did-key-creator/node_modules/@eslint/eslintrc/lib/config-array-factory.js not supported. .
......
```

I actually did reference did-key-creator from  https://github.com/ceramicnetwork/js-did/blob/main/packages/key-did-resolver/README.md and it was used in the tests for 
https://github.com/decentralized-identity/did-jwt/blob/master/src/__tests__/JWT.test.ts
nnd this in devDependencies https://github.com/decentralized-identity/did-jwt/blob/master/package.json
and CJS vs ESM was a problem in https://github.com/bshambaugh/nist-weierstrauss/issues/2

so in short, @ukstv might be interested in helping address this problem. 

Good catch addressing the ###  __snapshots__ and ### __SNAPSHOTS__ in this repository. I'm unsure. Is this an urgent enough problem that I should try to get that out before this PR is even complete?

**Comment by bshambaugh** [comment]
Status update. I'm still fidgeting with CJS and ESM. Bear with me as the migration to ESM has been compared to Python 2 vs 3.
I'm still learning.  

https://www.claritician.com/typescript-lib-vs-target-what-s-the-difference (comparing target and lib properties in tsconfig)
https://gist.github.com/bshambaugh/12dacbf186a3009fc5ccb28743878669 (tsc help docs)
